ksp_plugin/flight_plan.cpp

```diff
@@ -368,12 +377,21 @@ Status FlightPlan::BurnSegment(
 Status FlightPlan::CoastSegment(
     Instant const& desired_final_time,
     not_null<DiscreteTrajectory<Barycentric>*> const segment) {
-  return ephemeris_->FlowWithAdaptiveStep(
-                         segment,
-                         Ephemeris<Barycentric>::NoIntrinsicAcceleration,
-                         desired_final_time,
-                         adaptive_step_parameters_,
-                         max_ephemeris_steps_per_frame);
+  Instant const starting_time = segment->back().time;
+
+  // Don't try to integrate if the entire segment is not covered by the
+  // ephemeris yet.  Request a reanimation instead.
+  if (ephemeris_->t_min() <= starting_time) {
+    return ephemeris_->FlowWithAdaptiveStep(
+        segment,
+        Ephemeris<Barycentric>::NoIntrinsicAcceleration,
+        desired_final_time,
+        adaptive_step_parameters_,
+        max_ephemeris_steps_per_frame);
+  } else {
+    ephemeris_->RequestReanimation(starting_time);
+    return Status::OK;
```

Comment:
This is going to break all the invariants: we have a segment that has not been flowed, but we say it is OK. If we eventually hit a segment whose starting time is after t_min, we are going to try integrating that, and assuming this doesn’t crash, we will have a missing segment followed by a normal segment, which is not how anomaly works in the flight plan.
This should return an erroneous status so that the entire flight plan becomes anomalous.
This should probably also have a unit test.

Reply:
Done, PTAL.